dask/dataframe/shuffle.py

```diff
@@ -211,9 +211,15 @@ def set_index(
     upsample: float = 1.0,
     divisions: Sequence | None = None,
     partition_size: float = 128e6,
+    sort: bool = True,
     **kwargs,
 ) -> DataFrame:
     """See _Frame.set_index for docstring"""
+    if not sort:
+        df.divisions = tuple(None for _ in df.divisions)
+
+        return df.map_partitions(M.set_index, index, align_dataframes=False)
```

Comment:
Let's forward any kwargs that `pandas.DataFrame.set_index` supports too (e.g. `drop`) to the `set_index` call that happens on each partition

Reply:
https://github.com/dask/dask/pull/10342/commits/9199d1b7659d1507b5ef2120fa6f4880b1a71904

Reply:
Good catch, @jrbourbeau!